src/common/TrackedOp.h

```diff
@@ -386,4 +278,155 @@ class TrackedOp : public boost::intrusive::list_base_hook<> {
 };
 
 
+class OpTracker {
+  struct ShardedTrackingData {
+    mutable Mutex ops_in_flight_lock_sharded;
+    TrackedOp::tracked_op_list_t ops_in_flight_sharded;
+    explicit ShardedTrackingData(string lock_name):
+      ops_in_flight_lock_sharded(lock_name.c_str()) {
+    }
+  };
+
+  friend class OpHistory;
+  std::atomic<int64_t> seq = { 0 };
+  // preallocate space for 32 shards because of the default
+  // value for osd_num_op_tracker_shard
```

Comment:
Yup, this is clumsy. TODO:
   1) consider making the `OpTracker` a template to expose the parameter outside.
   2) maybe introduce to `md_config_t` a `constexp`r-capable getter for option's default value. This *might* be possible as with C++17 we've gotten  `std::variant` that makes use of `constexpr`.